lib/cocoapods/installer/xcode/multi_pods_project_generator.rb

```diff
@@ -0,0 +1,60 @@
+module Pod
+  class Installer
+    class Xcode
+      # The {MultiPodsProjectGenerator} handles generation of the 'Pods/Pods.xcodeproj' and Xcode projects
+      # for every {PodTarget}. All Pod Target projects are nested under the 'Pods.xcodeproj'.
+      #
+      class MultiPodsProjectGenerator < PodsProjectGenerator
+        # Generates `Pods/Pods.xcodeproj` and all pod target subprojects.
+        #
+        # @return [PodsProjectGeneratorResult]
+        #
+        def generate!
+          container_project_path = sandbox.project_path
+          build_configurations = analysis_result.all_user_build_configurations
+          platforms = aggregate_targets.map(&:platform)
+          object_version = aggregate_targets.map(&:user_project).compact.map { |p| p.object_version.to_i }.min
+          # Generate container Pods.xcodeproj.
+          container_project_generator = ProjectGenerator.new(sandbox, container_project_path, [],
+                                                             build_configurations, platforms, object_version,
+                                                             config.podfile_path)
+          container_project = container_project_generator.generate!
+
+          project_paths_by_pod_targets = pod_targets.group_by { |pod_target| sandbox.pod_target_project_path(pod_target) }
+          projects_by_pod_targets = Hash[project_paths_by_pod_targets.map do |project_path, pod_targets|
+            project = ProjectGenerator.new(sandbox, project_path,
+                                           pod_targets, build_configurations, platforms,
+                                           object_version, false, :pod_target_subproject => true).generate!
+            # Instead of saving every subproject to disk, we can optimize this by creating a temporary folder
+            # the file reference can use so that we only have to call `save` once for all projects.
+            project.path.mkdir unless File.exist?(project.path)
+            project_reference = container_project.add_subproject_reference(project, container_project.dependencies_group)
+            project_reference.name = project.path.basename('.*').to_s
```

Comment:
this should be done inside `add_subproject_reference`

Reply:
can it be a helper/class method so it can be used everywhere else too?

Reply:
suggested that `project_basename` be added to `Pod::Project`

Reply:
I didn't see until I scrolled further below

Reply:
Moved